src/main.rs

```diff
@@ -427,12 +443,10 @@ struct Chunk {
 struct FileReceiver {
     fname: String,
 
-    /// The file we’re writing to, if we have started writing.
-    ///
     /// We don’t open the file immediately so we don’t create a zero-sized file
     /// when a transfer fails. We only open the file after we have at least some
     /// data for it.
-    out_file: Option<File>,
+    file_created: bool,
```

Comment:
Why does this need to change? It was already acting like a bool (`Some` vs. `None`) and it was already preventing to re-open the file on every chunk receive. If you want to close the file after the receive is complete to limit the number of open file descriptors, we can change it to something with three states (like for sending), something like

```rust
enum FileReceiveState {
    /// We have received zero chunks so far.
    Pending,
    InProgress {
        file: File,
        pending: HashMap<u64, Chunk>,
        offset: u64,
    },
    Done,
}
```